lib/Sema/SemaBounds.cpp

```diff
@@ -932,6 +938,14 @@ namespace {
         return true;
       }
 
+      if (CK == CK_PointerBounds) {
```

Comment:
This is only needed if this is a dynamic_bounds_cast.   It is OK for the soure expression to not have a bound if it is an assume_bounds_cast operation.